lib/handleRequest.js

```diff
@@ -57,19 +57,60 @@ function handleRequest (req, res, params, context) {
   return
 }
 
-function jsonBody (request, reply) {
-  var body = ''
-  var req = request.req
-  req.on('error', onError)
+function jsonBody (request, reply, limit) {
+  const contentLength = Number.parseInt(request.headers['content-length'], 10)
+  if (contentLength > limit) {
+    reply.code(413).send(new Error('Request body is too large'))
+    return
+  }
+
+  const req = request.req
+  const chunks = []
```

Comment:
I’m confident that accumulating as a string is faster that accumulating in an array. Can you restore that?

You can call setEncoding(‘utf8’) to avoid relying on defaults.

Reply:
Or you can keep receiving buffers, but accumulate as String (it might be simpler to do the rest of the checks.

Reply:
Agree with @mcollina.

Reply:
After some benchmarking, I found that you're right @mcollina, accumulating as a string is faster. 

I also found that performance can be doubled by doing
`body += chunk.toString()` rather than `body += chunk`.

Reply:
Does it make sense to use `flatstr` after that step?

Reply:
no. flatstr is only useful when sending data out.